src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutPersister.cs

```diff
@@ -146,24 +200,14 @@ bool TryRemoveTimeoutEntity(Guid timeoutId, IDbConnection connection, out Timeou
                 using (var tx = session.BeginAmbientTransactionAware(IsolationLevel.ReadCommitted))
                 {
                     var te = session.Get<TimeoutEntity>(timeoutId);
+                    timeoutData = MapToTimeoutData(te);
```

Comment:
I think we could optimize the new TryRemove (without out data) in NHibernate to directly use the delete query, yes. I didn't do that to minimize changes on the code for the hotfixes. But im +1 to do that on the latest version.